packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts

```diff
@@ -318,60 +396,84 @@ export function applyConstraint(state: FuzzTestState, constraint: Constraint) {
+				break;
+			case "":
+				assert(pathStep.index !== undefined);
+				currentNode = (currentNode as ArrayChildren).at(pathStep.index) as TreeNode;
+				break;
+			case "arrayChildren": {
+				const arrayChildren =
+					(currentNode as FuzzNode).arrayChildren ??
+					fail(`Unexpected field type: ${pathStep.field}`);
+
+				currentNode = arrayChildren;
+				break;
+			}
+
+			case "optionalChild": {
+				const optionalChild =
+					(currentNode as FuzzNode).optionalChild ??
+					fail(`Unexpected field type: ${pathStep.field}`);
+				currentNode = optionalChild;
+				break;
+			}
+			case "requiredChild": {
+				const requiredChild =
+					(currentNode as FuzzNode).requiredChild ??
+					fail(`Unexpected field type: ${pathStep.field}`);
+				currentNode = requiredChild;
+				break;
+			}
 			default:
-				fail("Unknown field key");
+				fail(`Unexpected field type: ${pathStep.field}`);
 		}
 	}
-}
 
-function navigateToNode(tree: FuzzView, path: DownPath): FlexTreeNode {
-	const nodeSchema = tree.currentSchema;
-	const rootField = tree.flexTree;
-	const finalLocation = path.reduce<{
-		field: FlexTreeField;
-		containedNode: FlexTreeNode;
-	}>(
-		({ containedNode }, nextStep) => {
-			const childField = containedNode?.tryGetField(nextStep.field);
-			// Checking "=== true" causes tsc to fail to typecheck, as it is no longer able to narrow according
-			// to the .is typeguard.
-			/* eslint-disable @typescript-eslint/strict-boolean-expressions */
-			if (childField?.is(nodeSchema.objectNodeFieldsObject.sequenceChildren)) {
-				assert(nextStep.index !== undefined);
-				return {
-					field: childField,
-					containedNode: childField.at(nextStep.index) ?? fail("Index out of bounds."),
-				};
-			} else if (
-				// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-				childField?.is(nodeSchema.objectNodeFieldsObject.optionalChild) ||
-				childField?.is(nodeSchema.objectNodeFieldsObject.requiredChild)
-			) {
-				return {
-					field: childField,
-					containedNode: childField.content ?? fail("Missing child"),
-				};
-			}
-			/* eslint-enable @typescript-eslint/strict-boolean-expressions */
+	return currentNode;
+}
 
-			fail(`Unexpected field type: ${childField?.key}`);
-		},
-		{ field: rootField, containedNode: rootField.content ?? fail("Missing root") },
+function nodeSchemaForNodeType(nodeSchema: typeof FuzzNode, nodeType: string) {
+	assert(isObjectNodeSchema(nodeSchema));
+	const nodeSchemaField = nodeSchema.fields.get("requiredChild");
+	assert(nodeSchemaField !== undefined);
+	const allowedTypes = nodeSchemaField.allowedTypeSet;
+	const simpleNodeSchema = Array.from(allowedTypes).find(
+		(treeNodeSchema) => treeNodeSchema.identifier === nodeType,
 	);
+	const simpleSchema = simpleNodeSchema as unknown as new (dummy: unknown) => TreeNode;
+	return simpleSchema;
+}
 
-	return finalLocation.containedNode;
+function generateFuzzLeafNode(node: GeneratedFuzzNode, nodeSchema: typeof FuzzNode) {
```

Comment:
I'm finding this name misleading since it sometimes generate a non-leaf node. How about just `generateFuzzNode`?

Reply:
Updated name to `generateFuzzNode`